app/controllers/steps/conviction/conviction_length_controller.rb

```diff
@@ -0,0 +1,16 @@
+module Steps
+  module Conviction
+    class ConvictionLengthController < Steps::ConvictionStepController
+      def edit
+        @form_object = ConvictionLengthForm.new(
+          disclosure_check: current_disclosure_check,
+          conviction_length: current_disclosure_check.conviction_length
```

Comment:
`conviction_length_type` is missing. You can use build instead so it gets automatically the attributes from the form, refer to C100.